.github/workflows/native-full_marian-ubuntu.yml

```diff
@@ -71,12 +45,39 @@ jobs:
         sudo sh -c "echo deb https://apt.repos.intel.com/mkl all main > /etc/apt/sources.list.d/intel-mkl.list"
         sudo apt-get update -o Dir::Etc::sourcelist="/etc/apt/sources.list.d/intel-mkl.list"
         sudo apt-get install -y --no-install-recommends intel-mkl-64bit-2020.0-088
-      if: matrix.cpu == true
+      if: matrix.cpu == 'ON'
 
     # The script simplifies installation of different versions of CUDA
     - name: Install CUDA
       run: ./3rd_party/marian-dev/scripts/ci/install_cuda_ubuntu.sh ${{ matrix.cuda }}
-      if: matrix.gpu == true
+      if: matrix.gpu == 'ON'
+
+    # Lifted from https://cristianadam.eu/20200113/speeding-up-c-plus-plus-github-actions-using-ccache/
+    - name: Prepare ccache timestamp
+      id: ccache_cache_timestamp
+      shell: cmake -P {0}
+      run: |
+        string(TIMESTAMP current_date "%Y-%m-%d-%H;%M;%S" UTC)
+        message("::set-output name=timestamp::${current_date}")
+
+    - name: Setup ccache environment variables
+      run: | 
+        # Hash compiler content, mtime + size is modded everytime, so.
+        echo "CCACHE_COMPILERCHECK=content" >> $GITHUB_ENV 
+        echo "CCACHE_BASE_DIR=${{ github.workspace }}" >> $GITHUB_ENV
+        echo "CCACHE_DIR=${{ github.workspace }}/.ccache" >> $GITHUB_ENV
+        echo "CCACHE_COMPRESS=true" >> $GITHUB_ENV
+        echo "CCACHE_COMPRESSLEVEL=6" >> $GITHUB_ENV
+        echo "CCACHE_MAXSIZE=2G" >> $GITHUB_ENV
+
+    - name: Cache-op for build-cache through ccache 
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.CCACHE_DIR }}
+        key: ${{ matrix.name }}-ccache-${{ steps.ccache_cache_timestamp.outputs.timestamp }}
```

Comment:
@XapaJIaMnu any way I can get a command output which gives me a machine signature that is reproducible with cache that I can bake into key here?

Reply:
Or alternatively something which changes `-march=native` and makes it more general.